client.js

```diff
@@ -0,0 +1,254 @@
+        if (error.error_message.includes('_MIGRATE_')) {
+          const [_, nextDcId] = error.error_message.split('_MIGRATE_'); //eslint-disable-line
+          this.client.setDefaultDc(Number(nextDcId));
+          return await this.request(...args);
+        }
+        if (error.error_message === 'AUTH_RESTART') {
+          return await this.request(...args);
+        }
+        if (error.error_message.includes('FLOOD_WAIT')) {
+          const seconds = error.error_message.split('_WAIT_')[1];
+          throw new Error(`Our servers busy talking to Telegram. Please try again in ${seconds} seconds`);
+        }
+        throw new Error(error.error_message);
+      });
+  }
+
+  async checkIfLoggedIn(){
+    return await this.request('account.getAccountTTL', {});
+  }
+
+  async startLogin({mobile}){
+    const { phone_code_hash } = await this.request(
+      'auth.sendCode',
+      {
+        phone_number: mobile,
+        settings: {
+          _: 'codeSettings',
+        }
+      },
+    );
+    return { phoneCodeHash: phone_code_hash };
+  }
+
+  async completeLogin({ mobile, phoneCode, phoneCodeHash }) {
+    const response = await this.request(
+      'auth.signIn',
+      {
+        phone_code: phoneCode,
+        phone_number: mobile,
+        phone_code_hash: phoneCodeHash,
+      },
+    );
+
+    if (response._ === 'auth.authorizationSignUpRequired') {
+      const signUpRequiredMessage = `
+        There is no Telegram account associated with ${mobile}.
+        Please download the Telegram app on your mobile device and sign up, then try again.
+      `;
+      throw new Error(signUpRequiredMessage);
+    }
+    return response;
+  }
+
+  async loadMessagesFromChannel({ channelId, accessHash, earliestMessageId, limit }) {
+    const { messages, users } = await this.request(
+      'messages.getHistory',
+      {
+        peer: {
+          _: 'inputPeerChannel',
+          channel_id: channelId,
+          access_hash: accessHash,
+        },
+        offset_id: earliestMessageId,
+        limit,
+      },
+    );
+
+    return addUsersToMessages({
+      users,
+      messages: messages.filter(message => message._ === 'message' && message.message),
+    });
+  }
+
+  async sendMessageToChannel({ message, channelId, accessHash }) {
+    const { updates, users } = await this.request(
+      'messages.sendMessage',
+      {
+        peer: {
+          _: 'inputPeerChannel',
+          channel_id: channelId,
+          access_hash: accessHash,
+        },
+        message,
+        random_id: Math.floor(Math.random() * 10000000000000000000),
+      },
+    );
+    return getFormedMessagesFromUpdates({ updates, users, channelId });
+  }
+
+  async createChannel({ title, about }) {
+    const { chats } = await this.request(
+      'channels.createChannel',
+      {
+        flags: 2,
+        title,
+        about,
+      }
+    );
+
+    const chat = chats.find(chat => chat.title === title);
+    return {
```

Comment:
i like how this makes it clear what youre going to get back

Reply:
In this case we actually use the response